src/ui/public/agg_table/agg_table.js

```diff
@@ -93,6 +95,35 @@ uiModules
             formattedColumn.class = 'visualize-table-right';
           }
 
+          let isFirstValueNumeric =
+            table.rows && table.rows[0] && table.rows[0][i] && _.isNumber(table.rows[0][i].value);
+
+          if ((field && field.type === 'number') || isFirstValueNumeric) {
```

Comment:
We don't really these kind of complicated statements in ifs. Can you make it something like the `isFirstValueNumeric`, i.e. another variable with a great descriptor.